ClothesApp/Infrastructure/Repositories/ProductsRepository.cs

```diff
@@ -13,33 +14,75 @@ public ProductsRepository(ClothesAppContext dbContext)
         _dbContext = dbContext;
     }
 
+    public async Task<Product> Delete(Product product)
+    {
+        _dbContext.Remove(product);
+        await _dbContext.SaveChangesAsync();
+
+        return product;
+    }
+
     public async Task<bool> AnyProductOfBrandIdExists(long brandId)
     {
         return await _dbContext.Products.AnyAsync(p => p.BrandId == brandId);
     }
 
-    public Task<Product> Insert(Product entity)
+    public async Task<bool> DoesExist(string name)
+    {
+        return await _dbContext.Products.AnyAsync(p => p.Name == name);
+    }
+
+    public async Task<IList<Product>> FindByCondition(Expression<Func<Product, bool>> expression)
+    {
+        return await _dbContext.Products.Where(expression).ToListAsync();
+    }
+
+    public async Task UpdateRange(IList<Product> products)
+    {
+        _dbContext.Products.UpdateRange(products);
+        await _dbContext.SaveChangesAsync();
+    }
+
+    public async Task<bool> DoesExistRange(IList<long> ids)
+    {
+        var products = await _dbContext.Products.Where(p => ids.Contains(p.Id)).ToListAsync();
```

Comment:
You don't need to get list of products if you only need to count them, why not just CountAsync?